crates/subspace-service/src/sync_from_dsn/fast_sync.rs

```diff
@@ -507,14 +507,18 @@ where
 
             // TODO: add loop timeout
             let peer_candidates = loop {
-                let open_peers = network_service
-                    .connected_peers()
+                let connected_full_peers = self
+                    .sync_service
+                    .peers_info()
                     .await
-                    .expect("Network service must be available.");
+                    .expect("Network service must be available.")
```

Comment:
Is there a local invariant guarantees that? I'd return an error instead if not, it is not obvious why it is guaranteed here.